### ISSUE TITLE
Feature request: detect side-effects

### ISSUE THREAD

**Issue by espretto** [sweepline/eslint-plugin-unused-imports#70]
Hello 👋 

some of my modules have no exports but instead produce side-effects. would it be possible to annotate such modules w/ an eslint comment `/* eslint unused-imports: has-side-effects */` so their import statements don't get reported/removed?

**Comment by sweepline** [comment]
Why not just disable the rule in that file?
```
/* eslint-disable unused-imports/no-unused-imports */
```

**Comment by espretto** [comment]
This should work but only if applied to all import statements (call sites) across the codebase. It would be nice to do this once for every module having side-effects (definition sites).

**Comment by sweepline** [comment]
You are saying you have a file with no exports. Can't you just import it like `import "./path/to/js";`. It wont be reported unused then. As a second point, this is something that should be handled in the upstream package. This only splits out the rules made in `eslint` or `@typescript-eslint` and applies an autofix rule. So you should report it there.

If you would still like to discuss it further I would like an example of what you are trying to do.
